packages/flutter_adaptive_scaffold/test/adaptive_scaffold_test.dart

```diff
@@ -120,6 +126,48 @@ void main() {
     expect(tester.getTopLeft(sBody), const Offset(400, 0));
     expect(tester.getBottomRight(sBody), const Offset(800, 800));
   });
+
+  // The goal of this test is to run through each of the navigation elements
+  // and test whether tapping on that element will update the selected index
+  // globally
+  testWidgets('tapping navigation elements calls onSelectedIndexChange',
+      (WidgetTester tester) async {
+    // for each screen size there is a different navigational element that
+    // we want to test tapping to set the selected index
+    await Future.forEach(SimulatedLayout.values,
+        (SimulatedLayout region) async {
+      int selectedIndex = 0;
+      final MaterialApp app = region.app(initialIndex: selectedIndex);
+      await tester.binding.setSurfaceSize(region.size);
+      await tester.pumpWidget(app);
+      await tester.pumpAndSettle();
+
+      // tap on the next icon
+      selectedIndex = (selectedIndex + 1) % TestScaffold.destinations.length;
```

Comment:
This change is arbitrary. Anything that updates the selected index here to not be equal to its previous value is fine.